wave_containers/include/wave/containers/impl/landmark_measurement_container.hpp

```diff
@@ -19,7 +19,11 @@ using boost::multi_index::tag;
  * holding measurements of type T. See `wave::internal::measurement_container`.
  */
 template <typename T>
-struct landmark_container {
+struct container_traits<::wave::LandmarkMeasurementContainer<T>> {
+    // Define to give MeasurementContainerBase access to its derived type
+    using MeasurementType = T;
```

Comment:
Yes. `T` does not persist outside of this definition. `MeasurementType` is defined so it can be used by `MeasurementContainerBase`, from outside of this definition.